WordPress/gutenberg
# Style Engine: add typography and color to frontend

## What?

Experimental PR to see if and how we can port backend features added in https://github.com/WordPress/gutenberg/pull/40332 to the frontend in relation to color and typography

## Why?
To unify the frontend and backend. 

It might not work out since styles and classnames aren't added in the same callback 🤷 

## How?

Adding color and typography CSS rules to the existing ruleset (spacing).

~The addition of a new method `getClassnames` to generate block classnames based on preset attribute values passed to the style engine something similar to `var:preset|color|burnedToast`~

## Testing Instructions

Insert a group block into the post editor and tweak the typography and color styles.

Here's some test HTML

<details>

<summary>Block code</summary>

```html

<div class="wp-block-group has-secondary-background-color has-text-color has-background" style="font-style:normal;color:#dd1d1d;margin-top:58px;margin-bottom:58px;padding-top:45px;padding-right:45px;padding-bottom:45px;padding-left:45px;font-size:2.08rem;font-weight:500;letter-spacing:23px;line-height:2;text-transform:uppercase">
<p>I love chocolate raisins</p>
</div>


```

</details>

Check that the block appears as it should in the editor and the frontend.



### Unit tests

```bash
# style engine tests
npm run test-unit packages/style-engine/src/test/index.js 

# global styles output
npm run test-unit components/global-styles/test/use-global-styles-output.js
```
More to come
